internal/services/appconfiguration/app_configuration_resource.go

```diff
@@ -434,6 +488,61 @@ func resourceAppConfigurationUpdate(d *pluginsdk.ResourceData, meta interface{})
 		return fmt.Errorf("updating %s: %+v", *id, err)
 	}
 
+	if d.HasChange("replica") {
+		replicaClient := meta.(*clients.Client).AppConfiguration.ReplicasClient
+		operationsClient := meta.(*clients.Client).AppConfiguration.OperationsClient
+
+		// check if a replica has been removed from config and if so, delete it
+		deleteReplicaIds := make([]replicas.ReplicaId, 0)
+		oldReplicas, newReplicas := d.GetChange("replica")
+		for _, oldReplica := range oldReplicas.(*pluginsdk.Set).List() {
+			isRemoved := true
+			oldReplicaMap := oldReplica.(map[string]interface{})
+
+			for _, newReplica := range newReplicas.(*pluginsdk.Set).List() {
+				newReplicaMap := newReplica.(map[string]interface{})
+
+				if strings.EqualFold(oldReplicaMap["name"].(string), newReplicaMap["name"].(string)) && strings.EqualFold(oldReplicaMap["location"].(string), newReplicaMap["location"].(string)) {
```

Comment:
To ensure we're comparing these consistently and to prevent bugs we should normalise both location values since it isn't just the casing that can differ in these.

Reply:
fixed, thanks.